base-compat-batteries/src/Data/List/NonEmpty/Compat.hs

```diff
@@ -0,0 +1,78 @@
+{-# LANGUAGE CPP, NoImplicitPrelude, PackageImports #-}
+-- | This backports the modern "Data.Semigroup" interface back to
+-- @base-4.9@/GHC 8.0.
+module Data.List.NonEmpty.Compat (
+  -- * The type of non-empty streams
+    NonEmpty(..)
```

Comment:
Is the explicit export list necessary? That is, are the exports different on some version of `base`/`semigroups`? If they aren't, I'd prefer just re-exporting all of `Data.List.NonEmpty`.

Reply:
I see, I went with the same pattern as with other modules. Explicit export list also feels safer, travis + compiler check that there're all of these things (and as a bonus nicer haddocks :). I can change this, if you insist.

Reply:
> Explicit export list also feels safer, travis + compiler check that there're all of these things

Yes, but on the flip side, it's also entirely possible to _leave off_ something (and GHC certainly won't catch that).

If we are going to go down the route of explicit exports, I would feel much happier if there were some way to scrape this information directly from `base` itself. Doing this by hand feels far too error-prone for my liking.

Reply:
There's a PVP argument too: If we re-export modules as such, we have to depend on **minor** versions of dependency (i.e .`< base-4.11.1` or `semigroups <0.18.5`). This is because dependency may export new symbol (allowed in minor bump), and then `base-compat` should have minor bump as well!

> *Backwards compatible client specification.* Often a package maintainer wants to add to an API without breaking backwards compatibility, and in that case they can follow the rules of point 2, and increase only C. A client MAY specify that they are [insensitive to additions](https://wiki.haskell.org/Import_modules_properly) to the API by allowing a range of C values, e.g.  build-depends: base >= 2.1.1 && < 2.2.

(*insensitive to additions* is the important bit, PVP is very well thought!)

So I'd rather use explicit import lists (and *maybe* forget to export something, which is fixable by new release), then re-export modules as whole (which may cause confusion).

I'm happy to make a scraper (or first "comparator"): it's something trustees occasionally need too. (Let's have the "before GHC-8.6-rc1" as a deadline)

Reply:
Very well. If you volunteer to write this yourself, then I'm all for it :)

I've opened #49 for this issue.